### ISSUE TITLE
Bun + Nextjs + Docker: Incomplete .next build output in docker, but works fine locally

### ISSUE THREAD

**Issue by pdr-999** [oven-sh/bun#4795]
### What version of Bun is running?

1.0.0+822a00c4d508b54f650933a73ca5f4a3af9a7983

### What platform is your computer?

Linux 6.2.0-32-generic x86_64 x86_64

### What steps can reproduce the bug?

```
git clone git@github.com:pdr-999/bun-nextjs.git
```
```
docker compose up -d
```
```
docker compose logs app-bun
```

### What is the expected behavior?

bun-nextjs  | - ready started server on [::]:3000, url: http://localhost:3000

### What do you see instead?

bun-nextjs  | - warn You are using a non-standard "NODE_ENV" value in your environment. This creates inconsistencies in the project and is strongly advised against. Read more: https://nextjs.org/docs/messages/non-standard-node-env
bun-nextjs  | - ready started server on localhost:3000, url: http://localhost:3000
bun-nextjs  | ENOENT: No such file or directory
bun-nextjs  |    path: "/app/.next/BUILD_ID"
bun-nextjs  |  syscall: "open"
bun-nextjs  |    errno: -2

### Additional information

Docker version: Docker version 23.0.3, build 3e7cbfd

When running 
```
docker cp node-nextjs:/app ./app-bun
```
```
docker cp bun-nextjs:/app ./app-bun
```
.next contents of docker + nextjs + node
![image](https://github.com/oven-sh/bun/assets/107941092/9af4d143-ee78-4fb3-9fd0-7259548f5b89)

.next contents of docker + nextjs + bun
![image](https://github.com/oven-sh/bun/assets/107941092/37dd2176-621b-4fde-89a9-75e326cb03e6)

tldr:
Running 
```
bun install && bun run build && bun run start
```
- Incomplete in docker
- Complete locally

**Comment by osflash** [comment]
the command `bun run build` was not executed.

**Comment by marcfrankel** [comment]
I believe I am having a similar issue

**Comment by nacimgoura** [comment]
Same problem, `bun run build` doesn't work correctly inside docker

**Comment by doguyilmaz** [comment]
Same issue, nextjs + docker build fails.
Probably memory issue but could not find any equivalent for --max-old-space-size in Node.js yet.
(While running locally, there is no issue with build. v1.0.0)

![image](https://github.com/oven-sh/bun/assets/52458408/9e44c547-c73c-49b1-9459-a84dce220248)

**Comment by michaelangeloio** [comment]
Why create the build in docker? Why not build it outside of docker, copy it, and then just `bun run` in docker?

**Comment by marcfrankel** [comment]
In my specific instance, I am trying to get Next.js 13 to work with Bun on Railway. The build seems to complete successfully with no errors (although the next js file structure output is strangely missing from the console. Could be a red herring tho). However, the problem arises when the command bun run start is executed as I get the following issue:

```
$ next start

- ready started server on localhost:7988, url: http://localhost:7988/

ENOENT: No such file or directory

path: "/app/.next/BUILD_ID"

syscall: "open"

errno: -2
```

**Comment by pdr-999** [comment]
> the command `bun run build` was not executed.

In my case it did execute successfully but the output was incomplete, from this snippet of 
`docker compose build app-bun --progress=plain --no-cache` logs, it finishes before generating any pages. The output is usually `Generating static pages (4/4)`, `Finalizing page optimization.....` after `Creating an optimized production build`, etc, instead it finishes.
```
#9 [5/5] RUN bun run build
#9 0.561 $ next build
#9 0.590 - warn You are using a non-standard "NODE_ENV" value in your environment. This creates inconsistencies in the project and is strongly advised against. Read more: https://nextjs.org/docs/messages/non-standard-node-env
#9 1.229 Attention: Next.js now collects completely anonymous telemetry regarding usage.
#9 1.229 This information is used to shape Next.js' roadmap and prioritize features.
#9 1.229 You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
#9 1.229 https://nextjs.org/telemetry
#9 1.229 
#9 1.378 - info Creating an optimized production build...
#9 DONE 8.3s
```
> Why create the build in docker? Why not build it outside of docker, copy it, and then just `bun run` in docker?

Most Dockerfile examples with node builds in docker, 
https://github.com/vercel/next.js/blob/canary/examples/with-docker/Dockerfile
so I tried replacing node with bun.

Installing outside of docker defeats the purpose of using docker in a server no?

But shouldn't it have the same output either way?

**Comment by yicru** [comment]
Facing the same issue here.
Managed to bypass it by using a Node image in the build stage.

```Dockerfile
# syntax=docker/dockerfile:1

ARG BUN_VERSION=1.0.0

FROM oven/bun:${BUN_VERSION} as base

WORKDIR /app

ENV NODE_ENV=production

FROM base as deps

COPY package.json bun.lockb ./
RUN bun install

FROM node:18 as build

WORKDIR /app

COPY --from=deps /app/node_modules ./node_modules
COPY . .
RUN npm run build -- --no-lint

FROM base as final

USER bun

COPY --from=build /app/package.json ./package.json
COPY --from=build /app/node_modules ./node_modules
COPY --from=build /app/.next ./.next
COPY --from=build /app/public ./public

EXPOSE 3000

CMD ["bun", "run", "start"]
```

**Comment by AnatoliyKhaulin** [comment]
> Facing the same issue here. Managed to bypass it by using a Node image in the build stage.
> 
> ```dockerfile
> # syntax=docker/dockerfile:1
> 
> ARG BUN_VERSION=1.0.0
> 
> FROM oven/bun:${BUN_VERSION} as base
> 
> WORKDIR /app
> 
> ENV NODE_ENV=production
> 
> FROM base as deps
> 
> COPY package.json bun.lockb ./
> RUN bun install
> 
> FROM node:18 as build
> 
> WORKDIR /app
> 
> COPY --from=deps /app/node_modules ./node_modules
> COPY . .
> RUN npm run build -- --no-lint
> 
> FROM base as final
> 
> USER bun
> 
> COPY --from=build /app/package.json ./package.json
> COPY --from=build /app/node_modules ./node_modules
> COPY --from=build /app/.next ./.next
> COPY --from=build /app/public ./public
> 
> EXPOSE 3000
> 
> CMD ["bun", "run", "start"]
> ```

<img width="637" alt="Screenshot 2023-09-15 at 10 35 42" src="https://github.com/oven-sh/bun/assets/9891873/c4bc6694-7e98-41a0-b56f-6e08e536a148">

**Comment by aak-lear** [comment]
Same issue

next.js build with bun in Docker fails with following error:
script "build" exited with code 9 (SIGKILL)

**Comment by ericklarsen** [comment]
Same issue here

**Comment by nuntjw** [comment]
same issue as @aak-lear. any update?

**Comment by kaumac** [comment]
Also having this issue running `bun --bun run next build`

**Comment by akarabach** [comment]
the same issue for me

**Comment by iv-stpn** [comment]
I had a similar issue. The only solution that worked for me was to use a node image as base and install bun on it.

Instead of using:

```
FROM oven/bun:1.2.14
```

Using:

```
FROM node:22-alpine AS build

RUN apk add --no-cache curl bash

RUN curl -fsSL https://bun.sh/install | bash -s "bun-v1.2.14"
```

Hope it helps someone :)

**Comment by lucasdu4rte** [comment]
I used this Dockerfile, and it works for me:
```Dockerfile
# use the official Bun image
# see all versions at https://hub.docker.com/r/oven/bun/tags
FROM oven/bun:1.2.14-alpine AS base

WORKDIR /usr/src/app

# install dependencies into temp directory
# this will cache them and speed up future builds
FROM base AS install
RUN mkdir -p /temp/dev
COPY package.json bun.lock /temp/dev/
RUN cd /temp/dev && bun install --frozen-lockfile

# install with --production (exclude devDependencies)
RUN mkdir -p /temp/prod
COPY package.json bun.lock /temp/prod/
RUN cd /temp/prod && bun install --frozen-lockfile --production

# install TypeScript as production dependency for Next.js
RUN cd /temp/prod && bun add typescript

# copy node_modules from temp directory
# then copy all (non-ignored) project files into the image
FROM base AS prerelease
COPY --from=install /temp/dev/node_modules node_modules
COPY . .

# [optional] tests & build
ENV NODE_ENV=production
RUN bun run build

# copy production dependencies and source code into final image
FROM base AS release
COPY --from=install /temp/prod/node_modules node_modules
COPY --from=prerelease /usr/src/app/.next ./.next
COPY --from=prerelease /usr/src/app/public ./public
COPY --from=prerelease /usr/src/app/package.json .
COPY --from=prerelease /usr/src/app/next.config.ts .
COPY --from=prerelease /usr/src/app/tsconfig.json .

# run the app
USER bun
EXPOSE 3000/tcp
ENTRYPOINT [ "bun", "run", "start" ]

```
